examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/Serialization.kt

```diff
@@ -0,0 +1,388 @@
+                    password = "securePassword"
+                )
+            )
+            app.login(credentials)
+            // :snippet-end:
+        }
+    }
+
+    @OptIn(ExperimentalRealmSerializerApi::class)
+    @Test
+    fun customUserDataExperimentalSerializerTest() {
+        // :snippet-start: custom-data-serializer
+        @Serializable
+        class UserCustomData(
+            val favoriteColor: String
+        )
+        // :snippet-end:
+
+        val app: App = App.create(FLEXIBLE_APP_ID)
+        runBlocking {
+            app.login(Credentials.anonymous(reuseExisting = false))
+            // :snippet-start: custom-user-data-experimental-serializer
+            val user = app.currentUser!!
+            // :remove-start:
+            val functionResponse = user.functions
+                .call<BsonDocument>("writeCustomUserData",
+                    mapOf("userId" to user.id, "favoriteColor" to "blue")
+                )
+            delay(10)
+            user.refreshCustomData()
+            // :remove-end:
+            val customUserData = user.customData<UserCustomData>()
+            assertEquals("blue", customUserData!!.favoriteColor)
+            // :snippet-end:
+            val deleteResponse = user.functions
+                .call<BsonDocument>("deleteCustomUserData")
+            assertEquals(deleteResponse["deletedCount"], BsonInt32(1))
+            user.logOut()
+        }
+    }
+
+    @OptIn(ExperimentalRealmSerializerApi::class)
+    @Test
+    fun customProfileDataExperimentalSerializerTest() {
+        // :snippet-start: user-profile-serializer
+        @Serializable
+        class UserProfile(
+            val email: String
+        )
+        // :snippet-end:
+
+        val app: App = App.create(FLEXIBLE_APP_ID)
+        val email = "my.email@example.com"
+        val password = getRandom()
+        val credentials = Credentials.emailPassword(email, password)
+        runBlocking {
+            app.emailPasswordAuth.registerUser(email, password)
+            app.login(credentials)
+            // :snippet-start: user-profile-experimental-serializer
+            val user = app.currentUser!!
+            val userProfile = user.profile<UserProfile>()
+
+            assertEquals(userProfile.email, "my.email@example.com")
+            // :snippet-end:
+            user.delete()
+        }
+    }
+
+    @OptIn(ExperimentalKBsonSerializerApi::class, ExperimentalRealmSerializerApi::class)
+    @Test
+    fun setEjsonEncoderInAppConfiguration() {
+        @Serializable
+        class Serialization_Frogger(
+            val name: String
+        )
+        // :snippet-start: set-custom-ejson-serializer-for-app-configuration
+        AppConfiguration.Builder(FLEXIBLE_APP_ID)
+            .ejson(
+                EJson(
+                    serializersModule = SerializersModule {
+                        contextual(Serialization_Frogger::class, Serialization_Frogger.serializer())
```

Comment:
```suggestion
                        contextual(Serialization_Frogger::class, FroggerSerializer)
```
Contextual serializers are handy when a built-in serializer does not exist, in this case using `Frogger.serializer()` means a builtin serializer for `Frogger` exists.

Reply:
Ahh, got it. I did a little more reading up on this - does the updated example illustrate this better?